src/devices/DCMotor/DCMotor.cs

```diff
@@ -41,58 +41,130 @@ public virtual void Dispose()
+        public static DCMotor Create(PwmChannel speedControlChannel)
         {
-            if (settings.UseEnableAsPwm)
-            {
-                if (settings.Pin0.HasValue && settings.Pin1.HasValue && settings.PwmChannel != null)
-                {
-                    return new DCMotor3Pin(
-                        settings.PwmChannel,
-                        settings.Pin0.Value,
-                        settings.Pin1.Value,
-                        settings.Controller);
-                }
-
-                throw new ArgumentException("When enable pin is used for PWM all pins and PwmChannel must be set.");
-            }
-            else
-            {
-                int? pin0 = settings.Pin0;
-                int? pin1 = settings.Pin1;
-
-                if (!pin0.HasValue && pin1.HasValue)
-                {
-                    pin0 = pin1;
-                    pin1 = null;
-                }
-
-                if (settings.PwmChannel != null)
-                {
-                    if (pin0.HasValue == pin1.HasValue)
-                    {
-                        throw new ArgumentException("Only one of Pin0 and Pin1 must be set when PWM is specified.");
-                    }
-
-                    return new DCMotor2PinNoEnable(
-                        settings.PwmChannel,
-                        pin0,
-                        settings.Controller);
-                }
-                else
-                {
-                    if (!pin0.HasValue)
-                    {
-                        // pin1 must be null now since we swap values earlier when pin0 == null
-                        throw new ArgumentException("Pin0 or Pin1 must be set when PWM is not specified.");
-                    }
-
-                    return new DCMotor2PinNoEnable(new SoftwarePwmChannel(pin0.Value, 50, 0.0, controller: settings.Controller), pin1, settings.Controller);
-                }
-            }
+            if (speedControlChannel == null)
+                throw new ArgumentNullException(nameof(speedControlChannel));
+
+            return new DCMotor2PinNoEnable(speedControlChannel, -1, null);
+        }
+
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in one direction.
+        /// </summary>
+        /// <param name="speedControlPin">Pin used to control the speed of the motor with software PWM</param>
+        /// <param name="controller">GPIO controller related to the pin</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(int speedControlPin, GpioController controller = null)
+        {
+            if (speedControlPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(speedControlPin));
+
+            controller = controller ?? new GpioController();
+            return new DCMotor2PinNoEnable(
+                new SoftwarePwmChannel(speedControlPin, 50, 0.0, controller: controller),
```

Comment:
Why did we pick 50 hertz? Is this documented as the frequency that needs to be used?

Reply:
it's a sane choice. All H-bridges I've got allow max of 50Hz. You never really should need more than that. I'd wait until someone complains before changing this since I expect it won't ever be an issue

Reply:
Given that this is one of the bindings that we will support, I don't think that we should base a sane choice based on the devices that you've got 😄. This should be configurable with a parameter given that there is no way to access this softwarepwmchannel after initializing from the DCMotor.

Reply:
I'll add a documentation for 50Hz but you can override the choice by passing `PwmChannel` directly.